docs/api.yaml

```diff
@@ -0,0 +1,844 @@
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+  /courses:
+    get:
+      tags:
+        - "Course"
+      summary: "Find courses with optionnal search criteria"
+      parameters:
+        - name: "id"
+          in: "query"
+          description: "Identifier of the course to find"
+          required: false
+          schema:
+            type: string
+        - name: offset
+          in: query
+          required: false
+          description: "Zero-based offset for pagination. Defaults to `0`."
+          schema:
+            type: integer
+        - name: limit
+          in: query
+          required: false
+          description: "Pagination limit, that is the maximum number of elements that can be retrieved. Defaults to `1000` and cannot exceed `1000`."
+          schema:
+            type: integer
+
+      operationId: "findCourses"
+      responses:
+        200:
+          description: "List of courses"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Course'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+    post:
+      tags:
+        - "Course"
+      summary: "Create new Courses"
+      description: ""
+      operationId: "createCourses"
+      requestBody:
+        description: "Collection of courses to create"
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/CreateCourse'
+      responses:
+        200:
+          description: "Collection of created courses"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Course'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /teachers:
+    get:
+      tags:
+        - "Teacher"
+      summary: "Find teachers with optionnal search criteria"
+      parameters:
+        - name: "id"
+          in: "query"
+          description: "Identifier of the teacher to find"
+          required: true
+          schema:
+            type: string
+        - name: from
```

Comment:
From and to filters are not needed for teachers. There aren't that many, let consumers handle that